src/Editor.js

```diff
@@ -455,6 +455,29 @@ const Editor = () => {
       }
     }
   });
+  const resizeLayerByExtreme = (baseSelectedLayer, extreme) => {
+    setState(current => ({
+      ...current,
+      doc: {
+        ...current.doc,
+        layers: current.doc.layers.map(layer => {
+          if (selection.has(layer.id)) {
+            const updatedLayer = layer;
+            if (extreme === "narrowest" || extreme === "widest") {
+              updatedLayer.x1 = baseSelectedLayer.x1;
+              updatedLayer.x2 = baseSelectedLayer.x2;
+            } else {
+              updatedLayer.y1 = baseSelectedLayer.y1;
+              updatedLayer.y2 = baseSelectedLayer.y2;
+            }
+            return updatedLayer;
+          }
+
+          return layer;
+        })
```

Comment:
This function is currently mutating the state, which we want to avoid ([React docs for state](https://reactjs.org/docs/react-component.html#state), [React docs about performance](https://reactjs.org/docs/optimizing-performance.html#the-power-of-not-mutating-data) and [an article giving a good explanation](https://www.freecodecamp.org/news/handling-state-in-react-four-immutable-approaches-to-consider-d1f5c00249d5/)).

This is happening because both `updatedLayer` and `layer` are references to the same object (`updatedLayer === layer`) — any changes made on `updatedLayer` are changes on `layer`.

To avoid mutating the state, we need to create a new object and copy the values across. To do that we can use the [spread operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax):

```js
const updatedLayer = { ...layer };
```